File: src/lib/types/category.ts
export interface Category {
	id: string;
	name: string;
	prefix: string | null;
	icon: string | null;
	sort_order: number;
	created_at: string;
}

export interface CreateCategoryInput {
	name: string;
	prefix: string | null;
	icon: string | null;
}

export interface CategoryWithCount {
	id: string;
	name: string;
	prefix: string | null;
	item_count: number;
}

File: src-tauri/src/models/category.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Category {
    pub id: String,
    pub name: String,
    pub prefix: Option<String>,
    pub icon: Option<String>,
    pub sort_order: i64,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateCategoryInput {
    pub name: String,
    pub prefix: Option<String>,
    pub icon: Option<String>,
}
